Library/Homebrew/os/mac.rb

```diff
@@ -210,5 +210,13 @@ def pkgutil_info(id)
     def mdfind_query(*ids)
       ids.map! { |id| "kMDItemCFBundleIdentifier == #{id}" }.join(" || ")
     end
+
+    # Avoid mandatory reinstalls of `pkg-config` upon OS upgrade.
+    sig { void }
+    def symlink_current_pkgconfig_directory
```

Comment:
I don't think it's that weird -- it's specific to macOS because that's where we want to provide `.pc` files to help find system libraries. We don't currently have a reliable set of system libraries that we need to ship `.pc` files for on Linux.

Reply:
@carlocab Yeh, but it's not a function of the OS itself, it's just OS-specific code. It should be a function living somewhere else that's just a no-op on non-MacOS.